qualtran/bloqs/swap_network.py

```diff
@@ -255,3 +257,105 @@ def _swz_small() -> SwapWithZero:
     import_line='from qualtran.bloqs.swap_network import SwapWithZero',
     examples=(_swz_small,),
 )
+
+
+@frozen
+class MultiplexedCSwap(UnaryIterationGate):
+    r"""Swaps the $l$-th register into an ancilla using unary iteration.
+
+    Applies the unitary which peforms
+    $$
+        U |l\rangle|\psi_0\rangle\cdots|\psi_l\rangle\cdots|\psi_n\rangle|\mathrm{junk}\rangle
+        \rightarrow
+        |l\rangle|\psi_0\rangle\cdots|\mathrm{junk}\rangle\cdots|\psi_n\rangle|\psi_l\rangle
+    $$
+    through a combination of unary iteration and CSwaps.
+
+    The cost should be $L n_b + L - 2 + n_c$, where $L$ is the
+    iteration length, $n_b$ is the bitsize of
+    the registers to swap, and $n_c$ is the number of controls.
+
+    Args:
+        selection_regs: Indexing `select` signature of type Tuple[`SelectionRegisters`, ...].
+            It also contains information about the iteration length of each selection register.
+        bitsize: The size of the registers we want to swap.
+        nth_gate: A function mapping the composite selection index to a single-qubit gate.
```

Comment:
nit: fix docstring

Reply:
Done. One thing I realized (as this was just copied from the ApplyGateToLth.. bloq) is that there (or at least in the tests you pass a Signature into control_regs. Is it recommended to use Signature in place of a Tuple[Registers...]?

Reply:
(I've just been using Signature as the return value of the Bloq.signature property.

Reply:
> Is it recommended to use Signature in place of a Tuple[Registers...]?

As a function argument, we should use `Signature` only to define the `signature()` property of a Bloq. At call sites, I think it can be a convenient way of initializing a bunch of registers so it's fine. A `Signature` is an `Iterable[Register]`, so any function expecting a `Iterable[Register]` would work fine if we pass it a `Signature`.